redux_todo/src/App.js

```diff
@@ -0,0 +1,128 @@
+import Box from "@mui/material/Box";
+import { styled } from "@mui/material/styles";
+import ArrowDown from "./Components/ArrowDown";
+import InputContainer from "./Components/InputContainer";
+import TodoItem from "./Components/TodoItem";
+import Footer from "./Components/Footer";
+import { useSelector, useDispatch } from "react-redux";
+import { useMemo, useState } from "react";
+import {
+  addTodo,
+  checkAll,
+  toggleDone,
+  edit,
+  replace,
+  deleteTodo,
+  clickAll,
+  clickActive,
+  clickCompleted,
+  clear,
+} from "./actions/index";
+
+const Heading = styled(Box)`
+  font-size: 100px;
+  font-weight: 500;
+  text-align: center;
+  margin: 5px;
+  color: rgba(175, 47, 47, 0.25);
+`;
+
+const Top = styled(Box)`
+  border: 1px solid grey;
+  width: 600px;
+  height: 60px;
+  display: flex;
+  margin: auto;
+`;
+
+function App() {
+  //get all from state
+  const all = useSelector((state) => state.todoReducers.all);
+
+  //get mode from state
+  const mode = useSelector((state) => state.todoReducers.mode);
+
+  //initial state for inputs
+  const [newToDo, setNewToDo] = useState("");
+  const [editToDo, setEditToDo] = useState("");
+
+  //to dispatch action
+  const dispatch = useDispatch();
+
+  //giving filtered array
+  const listToMap = useMemo(() => {
+    if (mode === "active") {
+      return [...all].filter((item) => item.done === false);
+    } else if (mode === "completed") {
+      return [...all].filter((item) => item.done === true);
+    } else {
+      return all;
+    }
+  }, [mode, all]);
+
+  //count of all list
+  const list = useMemo(() => all.length, [all]);
```

Comment:
create a separate count function which will be able to give you the count of all, active and completed depending on the mode. no need for 3 functions. and name it something like countTodos